### PR TITLE
use lego instead of certbot in rhui build

### PR DESCRIPTION
**Pull request by hopkiw**
* use lego instead of certbot
* add optional output path param to the getsecret task template
* make daily image builds an optional parameter

### CODE REVIEW

File: concourse/pipelines/linux-image-build.jsonnet [hunk 1]
```diff
@@ -145,49 +160,61 @@ local ELImgBuildJob(image, workflow) = imgbuildjob {
...
+      },
     },
   ],
 
   // Append var to Daisy build task
-  buildtask+: { vars+: [
-    'instance_service_account=rhui-builder@rhel-infra.google.com.iam.gserviceaccount.com',
-    'tls_cert_path=../../../../tls-output/rhui.crt',
-  ] },
+  buildtask: RHUIImgBuildTask(workflow, '((.:gcs-url))') {
```
Comment (Linskeyd) [inline concourse/pipelines/linux-image-build.jsonnet:210]: Not necessarily for this PR, but I noticed extra_tasks follows snake case and this is squished into one non-delimited string. Should we make the args consistent?

Comment (hopkiw) [reply concourse/pipelines/linux-image-build.jsonnet:210]: sorry, i don't quite follow what you mean

File: concourse/pipelines/linux-image-build.jsonnet [hunk 2]
```diff
@@ -145,49 +160,61 @@ local ELImgBuildJob(image, workflow) = imgbuildjob {
...
     {
-      task: 'get-acme-key-id',
-      config: gcp_secret_manager.get_secret {
-        secret_name: 'rhui-acme-kid',
-        inputs+: gcp_secret_manager.get_secret.outputs,
+      task: 'get-rhui-tls-key',
+      config: gcp_secret_manager.getsecrettask {
+        secret_name: 'rhui-tls-key',
+        project: rhui_project,
+        output_path: 'accounts/' + acme_server + '/' + acme_email + '/keys/' + acme_email + '.key',
```
[the remark this answers is not in the document]
Comment (hopkiw) [reply concourse/pipelines/linux-image-build.jsonnet:191]: this is more or less the standard string building method in jsonnet. we can use the stdlib function `std.format` if we think it's more readable (it's not strictly necessary for simple concatenation). this line would then be:
```suggestion
        output_path: 'accounts/%s/%s/keys/%s.key' % [ acme_server, acme_email, acme_email ],
```

[the remark this answers is not in the document]
Comment (Linskeyd) [reply concourse/pipelines/linux-image-build.jsonnet:191]: IMO the format method is more readable, but what I was more so hoping jsonnet would have is this type of thing (taken from javascript, called [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)):

`accounts/${acme_server}/${acme_email}/keys/${acme_email}.key`

**Pull request merged**